src/WebJobs.Extensions.DurableTask/LocalGrpcListener.cs

```
@@ -256,7 +256,33 @@ private OrchestrationStatus[] GetStatusesNotToOverride()
 
             public async override Task<P.RaiseEventResponse> RaiseEvent(P.RaiseEventRequest request, ServerCallContext context)
             {
-                await this.GetClient(context).RaiseEventAsync(request.InstanceId, request.Name, Raw(request.Input));
+                bool throwStatusExceptionsOnRaiseEvent = this.extension.Options.ThrowStatusExceptionsOnRaiseEvent ?? this.extension.DefaultDurabilityProvider.CheckStatusBeforeRaiseEvent;
+                try
+                {
+                    await this.GetClient(context).RaiseEventAsync(request.InstanceId, request.Name, Raw(request.Input));
+                }
+                catch (ArgumentException)
+                {
+                    if (throwStatusExceptionsOnRaiseEvent)
+                    {
+                        throw new RpcException(new Status(StatusCode.InvalidArgument, "The instance id is invalid."));
```

I think we want to throw this exception regardless of the setting. The setting only controls the case where there is a status mismatch.

---

Thanks, Chris! I'm a bit confused about the [description ](https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs#L228)of this setting —so it only controls the case where the instance status isn't found, for which we would throw an ArgumentException) [here,](https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs#L811)  For other cases like ArgumentNullException due to missing instanceId, eventName, or taskName, we’ll still always throw, right?

---

> it only controls the case where the instance status isn't found, for which we would throw an ArgumentException

Yes, this is correct.